{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio.Alphabet.IUPAC import ExtendedIUPACProtein\n",
    "from collections import Counter\n",
    "import numpy as np\n",
    "\n",
    "def counts1d(ar, values):\n",
    "    return np.bincount(ar, minlength=max(values) + 1)[values]\n",
    "\n",
    "def char_counts(ar, axis=None, chars=None, return_chars=False):\n",
    "    ar = np.asanyarray(ar, 'c')\n",
    "    out_dtype = ar.dtype\n",
    "    if not chars:\n",
    "        chars = np.unique(ar)\n",
    "    else:\n",
    "        chars = np.asanyarray(chars, ar.dtype)\n",
    "    numeric_ar = ar.view('uint8')\n",
    "    numeric_chars = chars.view('uint8')\n",
    "    count_slice = lambda slice: counts1d(slice, numeric_chars)\n",
    "    out = np.apply_along_axis(count_slice, axis, numeric_ar)\n",
    "    if return_chars:\n",
    "        return out.T, numeric_chars.view(out_dtype)\n",
    "    else:\n",
    "        return out.T\n",
    "    \n",
    "def aln_profile(alignment, alphabet=None, return_chars=False):\n",
    "    return char_counts(alignment, chars=alphabet,\n",
    "                       axis=0, return_chars=return_chars)\n",
    "\n",
    "def aln_entropy(alignment):\n",
    "    prof = aln_profile(alignment)\n",
    "    pk = prof / np.sum(prof, axis=1).reshape(-1, 1)\n",
    "    \n",
    "    # log of the frequency of each character, but if the character\n",
    "    # is not counted, the value is 0, not -Inf or NaN\n",
    "    logpk = np.zeros_like(pk)\n",
    "    logpk[pk.nonzero()] = np.log(pk[pk.nonzero()])\n",
    "    \n",
    "    entropy = -np.sum(pk * logpk, axis=1)\n",
    "    return entropy\n",
    "\n",
    "def print_aln(alignment):\n",
    "    for rec in alignment:\n",
    "        print(rec.seq)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio.AlignIO import read as read_aln\n",
    "from Bio.Alphabet import generic_protein, generic_dna\n",
    "aln = read_aln('seq/refs.hmmalign.refine.afa', 'fasta', alphabet=generic_protein)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from matplotlib import pyplot as plt\n",
    "\n",
    "plt.plot(aln_entropy(aln))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def moving_window(x, size, f):\n",
    "    x = np.asarray(x)\n",
    "    out = np.empty(len(x))\n",
    "    for i in range(len(x)):\n",
    "        start = i\n",
    "        stop  = min(len(x), i + size)\n",
    "        out[i] = f(x[start:stop])\n",
    "    return out\n",
    "\n",
    "def smooth(x, bw):\n",
    "    return moving_window(x, bw, f = np.mean)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "ax = plt.subplot(111)\n",
    "ax.plot(smooth(aln_entropy(aln), 5))\n",
    "ax.axvspan(350, 375, color='k', alpha=0.1)\n",
    "ax.axvspan(515, 535, color='k', alpha=0.1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "trace = aln_entropy(aln)\n",
    "\n",
    "ax = plt.subplot(111)\n",
    "ax.plot(trace)\n",
    "\n",
    "regions = [(360, 369),  # Perfect AA conservation\n",
    "           (442, 448),  # Perfect, but short (maybe 1 more 5' base)\n",
    "           (493, 500),  # Imperfect and short\n",
    "           (520, 531),  # Perfect AA conservation\n",
    "          ]\n",
    "\n",
    "min_max = lambda lst: (min(lst), max(lst))\n",
    "minimum, maximum = min_max([coord for reg in regions\n",
    "                                  for coord in reg])\n",
    "ax.set_xlim(minimum - 10, maximum + 10)\n",
    "\n",
    "for start, stop in regions:\n",
    "    ax.axvspan(start, stop, color='g', alpha=0.15)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "consensus(prot[:,440:450])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from collections import Counter\n",
    "\n",
    "def consensus(alignment):\n",
    "    prof, chars = aln_profile(alignment, return_chars=True)\n",
    "    return chars[np.argmax(prof, axis=1)].astype(str)\n",
    "    \n",
    "\n",
    "for start, stop in regions:\n",
    "    print(\"\".join(consensus(aln[:,start:stop])))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "reverse_translation_table = \\\n",
    "{'F': 'TTY',\n",
    " 'L': 'YTN',\n",
    " 'I': 'ATH',\n",
    " 'M': 'ATG',\n",
    " 'V': 'GTN',\n",
    " 'S': 'WSN',\n",
    " 'P': 'CCN',\n",
    " 'T': 'ACN',\n",
    " 'A': 'GCN',\n",
    " 'Y': 'TAY',\n",
    " 'H': 'CAY',\n",
    " 'Q': 'CAR',\n",
    " 'N': 'AAY',\n",
    " 'K': 'AAR',\n",
    " 'D': 'GAY',\n",
    " 'E': 'GAR',\n",
    " 'C': 'TGY',\n",
    " 'W': 'TGG',\n",
    " 'R': 'MGN',\n",
    " 'G': 'GGN',\n",
    "}\n",
    "\n",
    "def back_translate(prot_seq):\n",
    "    out = []\n",
    "    for res in prot_seq:\n",
    "        out.append(reverse_translation_table[res])\n",
    "    return \"\".join(out)\n",
    "\n",
    "back_translate(\"FLIMVSPTAYHQNKDECWRG\")\n",
    "# TTYYTNATHATGGTNWSNCCNACNGCNTAYCAYCARAAYAARGAYGARTGYTGGMGNGGN"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio.Seq import Seq\n",
    "from Bio.Alphabet import IUPAC\n",
    "\n",
    "for motif in (consensus(aln[:,start:stop]) for start, stop in regions):\n",
    "    seq = Seq(back_translate(motif), alphabet=IUPAC.ambiguous_dna)\n",
    "    print(\"{!s: <40}{!s: <40}\".format(seq, seq.reverse_complement()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio.AlignIO import read as read_aln\n",
    "from Bio.Alphabet import generic_protein, generic_dna, Gapped\n",
    "from Bio.Alphabet.IUPAC import protein, unambiguous_dna\n",
    "import numpy as np\n",
    "\n",
    "prot = read_aln('seq/refs.mcra-hmmaln.auto-refine.afa', 'fasta',\n",
    "                alphabet=Gapped(protein))\n",
    "nucl = read_aln('seq/refs.mcra-hmmaln.auto-refine.afn', 'fasta',\n",
    "                alphabet=Gapped(unambiguous_dna))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "plt.plot(smooth(aln_entropy(nucl), 20))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def nucl_freq_plot(xs, *args, ax=None, colors=None, **kwargs):\n",
    "    total = np.sum(xs, axis=0)\n",
    "    tally = np.zeros_like(total)\n",
    "    pos = range(len(total))\n",
    "    if not ax:\n",
    "        ax = plt\n",
    "    if not colors:\n",
    "        colors = 'g', 'b', 'k', 'r', 'w'\n",
    "    for values, c in zip(xs, colors):\n",
    "        ax.bar(pos, values / total, bottom=tally / total, color=c)\n",
    "        tally += values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "fig = plt.figure(figsize=(13,6))\n",
    "\n",
    "for i, region in enumerate(regions, start=1):\n",
    "    ax = fig.add_subplot(2, 2, i)\n",
    "    nucl_coords = (region[0] * 3, (region[1] - 1) * 3)\n",
    "    nucl_freq_plot(aln_profile(nucl[:,nucl_coords[0]:nucl_coords[1]]).T, ax=ax)\n",
    "    ax.set_xlim(-5, 35)\n",
    "    ax.set_title(nucl_coords)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "reverse_translation_table = {\n",
    "    'F': 'TTY',\n",
    "    'L': 'YTN',\n",
    "    'I': 'ATH',\n",
    "    'M': 'ATG',\n",
    "    'V': 'GTN',\n",
    "    'S': 'WSN',\n",
    "    'P': 'CCN',\n",
    "    'T': 'ACN',\n",
    "    'A': 'GCN',\n",
    "    'Y': 'TAY',\n",
    "    'H': 'CAY',\n",
    "    'Q': 'CAR',\n",
    "    'N': 'AAY',\n",
    "    'K': 'AAR',\n",
    "    'D': 'GAY',\n",
    "    'E': 'GAR',\n",
    "    'C': 'TGY',\n",
    "    'W': 'TGG',\n",
    "    'R': 'MGN',\n",
    "    'G': 'GGN',\n",
    "}\n",
    "\n",
    "codon_degen_table = {\n",
    "    'F': 2,\n",
    "    'L': 8,\n",
    "    'I': 3,\n",
    "    'M': 1,\n",
    "    'V': 4,\n",
    "    'S': 16,\n",
    "    'P': 4,\n",
    "    'T': 4,\n",
    "    'A': 4,\n",
    "    'Y': 2,\n",
    "    'H': 2,\n",
    "    'Q': 2,\n",
    "    'N': 2,\n",
    "    'K': 2,\n",
    "    'D': 2,\n",
    "    'E': 2,\n",
    "    'C': 2,\n",
    "    'W': 1,\n",
    "    'R': 8,\n",
    "    'G': 4,\n",
    "}\n",
    "\n",
    "codon_degen_table_3_trunc = {\n",
    "    'F': 1,\n",
    "    'L': 2,\n",
    "    'I': 1,\n",
    "    'M': 1,\n",
    "    'V': 1,\n",
    "    'S': 4,\n",
    "    'P': 1,\n",
    "    'T': 1,\n",
    "    'A': 1,\n",
    "    'Y': 1,\n",
    "    'H': 1,\n",
    "    'Q': 1,\n",
    "    'N': 1,\n",
    "    'K': 1,\n",
    "    'D': 1,\n",
    "    'E': 1,\n",
    "    'C': 1,\n",
    "    'W': 1,\n",
    "    'R': 2,\n",
    "    'G': 1,\n",
    "}\n",
    "\n",
    "codon_degen_table_5_trunc = {\n",
    "    'F': 2,\n",
    "    'L': 4,\n",
    "    'I': 3,\n",
    "    'M': 1,\n",
    "    'V': 4,\n",
    "    'S': 8,\n",
    "    'P': 4,\n",
    "    'T': 4,\n",
    "    'A': 4,\n",
    "    'Y': 2,\n",
    "    'H': 2,\n",
    "    'Q': 2,\n",
    "    'N': 2,\n",
    "    'K': 2,\n",
    "    'D': 2,\n",
    "    'E': 2,\n",
    "    'C': 2,\n",
    "    'W': 1,\n",
    "    'R': 4,\n",
    "    'G': 4,\n",
    "}\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "aln_profile(prot[:,regions[2][0]:regions[2][1]], return_chars=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "motifs = []\n",
    "\n",
    "for start, stop in regions:\n",
    "    motifs.append(\"\".join(consensus(aln[:,start:stop])))\n",
    "    \n",
    "for motif in motifs:\n",
    "    for aa in motif:\n",
    "        print(\"{!s: <3}\".format(aa), sep=\"\", end=\"\")\n",
    "    print(\"\\n\", sep=\"\", end=\"\")\n",
    "    for aa in motif:\n",
    "        print(\"{!s: <3}\".format(codon_degen_table[aa]), sep=\"\", end=\"\")\n",
    "    print(\"\\n\", sep=\"\", end=\"\")\n",
    "    print(\"\\n\", sep=\"\", end=\"\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def calc_degen(motif, reverse=False):\n",
    "    tally = 1\n",
    "    for i, aa in enumerate(motif):\n",
    "        if reverse and (i == 0):\n",
    "            tally *= codon_degen_table_5_trunc[aa]\n",
    "        elif (not reverse) and (i == len(motif) - 1):\n",
    "            tally *= codon_degen_table_3_trunc[aa]\n",
    "        else:\n",
    "            tally *= codon_degen_table[aa]\n",
    "    return tally\n",
    "    \n",
    "\n",
    "def minimum_subseq_degen(motif, length, reverse=False):\n",
    "    hypotheses = []\n",
    "    for start in range(len(motif) - length):\n",
    "        hypothesis = motif[start:start+length]\n",
    "        hypotheses.append((hypothesis, calc_degen(hypothesis, reverse)))\n",
    "    return hypotheses\n",
    "\n",
    "minimum_subseq_degen('', 4, reverse=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "minimum_subseq_degen('DLQDQCG', 4, reverse=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "seq = Seq(back_translate('LQDQ'), alphabet=IUPAC.ambiguous_dna)\n",
    "print(\"{!s: <40}{!s: <40}\".format(seq, seq.reverse_complement()))"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.4.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
